### PR TITLE
Fix spelling via 'make codespell'

### PR DESCRIPTION
**Pull request by rockdaboot**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -86,15 +86,15 @@ This release brings a breaking change for custom trace API implementations. Some
 - `go.opentelemetry.io/otel/bridge/opencensus.NewMetricProducer` returns a `*MetricProducer` struct instead of the metric.Producer interface. (#4583)
 - The `TracerProvider` in `go.opentelemetry.io/otel/trace` now embeds the `go.opentelemetry.io/otel/trace/embedded.TracerProvider` type.
   This extends the `TracerProvider` interface and is is a breaking change for any existing implementation.
-  Implementors need to update their implementations based on what they want the default behavior of the interface to be.
+  Implementers need to update their implementations based on what they want the default behavior of the interface to be.
```
Comment (dmathieu) [inline CHANGELOG.md:89]: `Implementors` seems valid.
https://www.merriam-webster.com/dictionary/implement

Comment (pellared) [reply CHANGELOG.md:89]: Implementers is seems also good (personally I prefer it)

**Comment by pellared** [comment]
> We should figure out why the codespell action doesn't fail when it runs on CI.

This could be the reason [343 commits behind](https://github.com/rockdaboot/opentelemetry-go/compare/fix-spelling...open-telemetry%3Aopentelemetry-go%3Amain) open-telemetry/opentelemetry-go:main

**Comment by rockdaboot** [comment]
Sorry, my `main` branch was outdated (Monday morning without coffee).

**Pull request closed without merging**